apps/couchbase.go

```diff
@@ -0,0 +1,186 @@
+					},
+				},
+			},
+		},
+		Rules: []confgenerator.MultilineRule{
+			{
+				StateName: "start_state",
+				NextState: "cont",
+				Regex:     `^\[([^\s+:]*):`,
+			},
+			{
+				StateName: "cont",
+				NextState: "cont",
+				Regex:     `^(?!\[([^\s+:]*):).*$`,
+			},
+		},
+	}.Components(tag, lr.Type())...)
+
+	components = append(components,
+		confgenerator.LoggingProcessorModifyFields{
+			Fields: map[string]*confgenerator.ModifyField{
+				"severity": {
+					MoveFrom: "jsonPayload.level",
+					MapValues: map[string]string{
+						"debug": "DEBUG",
+						"info":  "INFO",
+						"warn":  "WARNING",
+						"error": "ERROR",
+					},
+					MapValuesExclusive: true,
+				},
+				InstrumentationSourceLabel: instrumentationSourceValue(lr.Type()),
+			},
+		}.Components(tag, lr.Type())...)
+	return components
+}
+
+// LoggingProcessorCouchbaseHTTPAccess is a struct that will generate the fluentbit components for the http access logs
+type LoggingProcessorCouchbaseHTTPAccess struct {
+	confgenerator.ConfigComponent           `yaml:",inline"`
+	confgenerator.LoggingReceiverFilesMixin `yaml:",inline" validate:"structonly"`
+}
+
+// Type returns the string for the couchbase http access logs
+func (lp LoggingProcessorCouchbaseHTTPAccess) Type() string {
+	return "couchbase_http_access"
+}
+
+// Components returns the fluentbit components for the http access logs of couchbase
+func (lp LoggingProcessorCouchbaseHTTPAccess) Components(tag string) []fluentbit.Component {
+	if len(lp.IncludePaths) == 0 {
+		lp.IncludePaths = []string{
+			"/opt/couchbase/var/lib/couchbase/logs/http_access.log",
+			"/opt/couchbase/var/lib/couchbase/logs/http_access_internal.log",
+		}
+	}
+	c := lp.LoggingReceiverFilesMixin.Components(tag)
+	c = append(c,
+		confgenerator.LoggingProcessorParseRegex{
+			Regex: `^(?<client_ip>[^ ]*) [^ ]* (?<user>[^ ]*) \[(?<timestamp>[^\]]*)\] "(?<method>\S+)(?: +(?<path>[^ ]*) +\S*)?" (?<status_code>[^ ]*) (?<response_size>[^ ]*) - (?<message>.*)$`,
+			ParserShared: confgenerator.ParserShared{
+				TimeKey:    "timestamp",
+				TimeFormat: `%d/%b/%Y:%H:%M:%S %z`,
+				Types: map[string]string{
+					"size": "integer",
+					"code": "integer",
+				},
+			},
+		}.Components(tag, "couchbase_http_access")...,
+	)
+	c = append(c, confgenerator.LoggingProcessorModifyFields{
+		Fields: map[string]*confgenerator.ModifyField{
+			InstrumentationSourceLabel: instrumentationSourceValue(lp.Type()),
+		},
+	}.Components(tag, lp.Type())...)
+	return c
+}
+
+// LoggingProcessorCouchbaseGOXDCR is a struct that iwll generate the fluentbit components for the goxdcr logs
+type LoggingProcessorCouchbaseGOXDCR struct {
+	confgenerator.ConfigComponent           `yaml:",inline"`
+	confgenerator.LoggingReceiverFilesMixin `yaml:",inline" validate:"structonly"`
+}
+
+// Type returns the type string for the cross datacenter logs of couchbase
+func (lg LoggingProcessorCouchbaseGOXDCR) Type() string {
+	return "couchbase_xdcr"
```

Comment:
This should probably be `couchbase_goxdcr`

Reply:
Should be addressed now 👍